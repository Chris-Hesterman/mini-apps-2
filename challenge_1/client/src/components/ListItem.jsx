import React from 'react';
import styled from 'styled-components';

const StyledListItem = styled.div`
  display: flex;
  flex-direction: column;
  background: rgba(230, 230, 230, 0.5);
  border-radius: 10px;
  padding: 5px;
`;

const StyledH4 = styled.h4`
  margin: 0 0 10px 0;
`;

const StyledPWrapper = styled.div`
  max-width: 60vw;
  overflow-x: hidden;
`;
const StyledP = styled.p`
  display: inline-block;
  margin: 0;
  text-indent: 2rem;
  max-width: 60vw;
  min-width: 60vw;
`;

const ListItem = (props) => {
  return (
    <StyledListItem>
      <StyledH4>Year: {props.event.date} ></StyledH4>
      <StyledPWrapper>
        <StyledP>{props.event.description}</StyledP>
      </StyledPWrapper>
    </StyledListItem>
  );
};

export default ListItem;
