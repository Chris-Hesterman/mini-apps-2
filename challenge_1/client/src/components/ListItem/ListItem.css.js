import styled from 'styled-components';

// ListItem.jsx styles --------
export const StyledListItem = styled.div`
  display: flex;
  flex-direction: column;
  background: rgba(230, 230, 230, 0.5);
  border-radius: 10px;
  padding: 5px;
`;

export const StyledItemH4 = styled.h4`
  margin: 0 0 10px 0;
`;

export const StyledItemPWrapper = styled.div`
  max-width: 60vw;
  overflow-x: hidden;
`;

export const StyledItemP = styled.p`
  display: inline-block;
  margin: 0;
  text-indent: 2rem;
  max-width: 60vw;
  min-width: 60vw;
`;
