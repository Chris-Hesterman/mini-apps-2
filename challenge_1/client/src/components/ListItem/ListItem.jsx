import React from 'react';
import styled from 'styled-components';
import {
  StyledListItem,
  StyledItemH4,
  StyledItemPWrapper,
  StyledItemP
} from './ListItem.css.js';

const ListItem = (props) => {
  return (
    <StyledListItem>
      <StyledItemH4>Year: {props.event.date}</StyledItemH4>
      <StyledItemPWrapper>
        <StyledItemP>{props.event.description}</StyledItemP>
      </StyledItemPWrapper>
    </StyledListItem>
  );
};

export default ListItem;
