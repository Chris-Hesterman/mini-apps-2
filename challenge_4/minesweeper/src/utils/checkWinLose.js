export const checkWin = (uncovered, mines) => {
  if (uncovered.size === 100 - mines.length) {
    return true;
  }

  return false;
};

export const checkLose = (squareNum, mines, flags) => {
  if (mines.includes(squareNum) && !flags.includes(squareNum)) {
    return true;
  }
  return false;
};
