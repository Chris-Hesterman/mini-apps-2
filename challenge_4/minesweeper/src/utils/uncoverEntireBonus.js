import checkAdjacent from './checkAdjacent';

export const uncoverEntireBonus = (number, uncoveredSquares, minesArray) => {
  let fullBonus = [];

  const helper = (num) => {
    let arr = buildAdjacent(num, uncoveredSquares, fullBonus);

    fullBonus = fullBonus.concat(arr);
    arr = arr.forEach((number) => {
      if (checkAdjacent(number, minesArray) === 0) {
        helper(number);
      }
    });
  };
  helper(number);

  return new Set(fullBonus);
};

export const buildAdjacent = (num, uncovered, bonus) => {
  let adjArray = [num - 10, num + 10];

  if (num % 10 !== 0 && num % 10 !== 1) {
    adjArray.push(num - 1, num + 1, num - 9, num + 9, num - 11, num + 11);
  }

  if (num % 10 === 1) {
    adjArray.push(num + 1, num - 9, num + 11);
  }

  if (num % 10 === 0) {
    adjArray.push(num + 9, num - 1, num - 11);
  }

  adjArray = adjArray.filter((number) => {
    if (uncovered.length) {
      if (
        number > 0 &&
        number < 101 &&
        uncovered.indexOf(number) === -1 &&
        !bonus.includes(number)
      ) {
        return number;
      }
    } else {
      if (number > 0 && number < 101) {
        return number;
      }
    }
  });

  return adjArray;
};
