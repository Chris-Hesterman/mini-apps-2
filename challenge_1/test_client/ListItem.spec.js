import React from 'react';
import PropTypes from 'prop-types';
import { checkProps } from '../testUtils/utils.js';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ListItem from '../client/src/components/ListItem/ListItem.jsx';
import { findByTestAttr } from '../testUtils/utils.js';

Enzyme.configure({ adapter: new Adapter() });

describe('ListItem component', () => {
  describe('Checking propTypes', () => {
    it('Should not throw a warning', () => {
      const expectedProps = {
        eventData: [{ one: 1 }, { two: 2 }, { three: 3 }, { four: 4 }],
        currentQuery: 'Test Query',
        event: { date: 23, description: 'test historical event' }
      };
      const propsError = checkProps(ListItem, expectedProps);

      expect(propsError).toBe(undefined);
    });
  });

  describe('Snapshot test', () => {
    const event = {
      date: '-297',
      description:
        'Demetrius Poliorcetes returns to Greece with the aim of becoming master of Macedonia. While Demetrius is in Greece, Lysimachus seizes his possessions in Asia Minor.',
      lang: 'en',
      category1: 'By place',
      category2: 'Greece',
      granularity: 'year'
    };

    it('should render correctly', () => {
      const tree = renderer.create(<ListItem event={event} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
