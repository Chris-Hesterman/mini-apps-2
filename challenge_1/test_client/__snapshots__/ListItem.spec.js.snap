// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component Snapshot test should render correctly 1`] = `
<div
  className="ListItemcss__StyledListItem-sc-1lk54j5-0 gBfNCw"
>
  <h4
    className="ListItemcss__StyledItemH4-sc-1lk54j5-1 eljxyj"
  >
    Year: 
    -297
  </h4>
  <div
    className="ListItemcss__StyledItemPWrapper-sc-1lk54j5-2 dhJpca"
  >
    <p
      className="ListItemcss__StyledItemP-sc-1lk54j5-3 cItsBx"
    >
      Demetrius Poliorcetes returns to Greece with the aim of becoming master of Macedonia. While Demetrius is in Greece, Lysimachus seizes his possessions in Asia Minor.
    </p>
  </div>
</div>
`;
